### PR TITLE
[api/labels] Initialize nil dest labels during addition.

### PR DESCRIPTION
**Pull request by dilyevsky**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

Fixes panic on setting `GCPMachine.AdditionalLabels` map when `GCPCluster.AdditionalLabels` is not set.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*: n/a

**Special notes for your reviewer**: n/a

**TODOs**:

- [x] squashed commits
- [x] includes documentation (already up-to-date docstring).
- [x] adds unit tests

**Release note**:

```release-note
Fixes panic on setting `GCPMachine.AdditionalLabels` map when `GCPCluster.AdditionalLabels` is not set.
```

### CODE REVIEW

**Comment by dilyevsky** [comment]
@vincepri @gab-satchi fixed fmt issue if you have time to take a look

**Review by cpanato — COMMENTED** [review]
/ok-to-test

**Comment by dilyevsky** [comment]
gofmt accedientally butchered codegen'ed file 🤦 ... @cpanato can you try again pls

**Comment by dilyevsky** [comment]
actually nvm looks like it's retrying on its down, lets see if this works now

**Comment by dilyevsky** [comment]
hm for some reason it fails on a file that's hasn't been changed in 5 months:

```
./hack/boilerplate/boilerplate.py -v
File /home/dilyevsky/projects/cluster-api-provider-gcp/tilt_modules/cert_manager/test/test.sh smaller than reference (5 < 14)
```

**Comment by dilyevsky** [comment]
hm the boilerplate.py now breaks on autogen'ed .go files @cpanato any pointers how to address this (i maybe lacking context)?

**Comment by cpanato** [comment]
> hm for some reason it fails on a file that's hasn't been changed in 5 months:
> 
> ```
> ./hack/boilerplate/boilerplate.py -v
> File /home/dilyevsky/projects/cluster-api-provider-gcp/tilt_modules/cert_manager/test/test.sh smaller than reference (5 < 14)
> ```

sorry for the trouble that might cause, we enable a new missing check and some things in the repo were not compliant, as soon this https://github.com/sigstore/cosign/issues/974 gets merged you can rebase your pr the the issue will be fixed

File: api/v1alpha3/zz_generated.deepcopy.go [hunk 1]
```diff
@@ -1,3 +1,4 @@
+//go:build !ignore_autogenerated
```
Comment (cpanato) [inline api/v1alpha3/zz_generated.deepcopy.go:1]: are you using go 1.17?

Comment (cpanato) [reply api/v1alpha3/zz_generated.deepcopy.go:1]: please use go 1.16.x for now

Comment (dilyevsky) [reply api/v1alpha3/zz_generated.deepcopy.go]: still adding those on `go version go1.16.8 linux/amd64`

Comment (dilyevsky) [reply api/v1alpha3/zz_generated.deepcopy.go]: I have reverted those manually.

**Review by cpanato — APPROVED** [review]
thanks, and sorry for the issues during the pr.
/lgtm
/approve

can you cherry pick this to the release-0.4 branch? if you dont have time i can do that as well

**Pull request merged**